src/maliput/base/distance_router.cc

```diff
@@ -0,0 +1,218 @@
+      return false;
+    }
+    if (routing_constraints.max_route_cost.has_value() &&
+        routing_constraints.max_route_cost.value() < ComputeCost(route)) {
+      return false;
+    }
+    return true;
+  };
+  std::vector<routing::Route> filtered_routes;
+  std::copy_if(routes.begin(), routes.end(), std::back_inserter(filtered_routes), filter);
+  return filtered_routes;
+}
+
+// Concatenates items in @p errors into just one string.
+//
+// When @p errors is empty an empty string computed.
+// Otherwise, the "Route has connectivity errors: <error> | <error> | ... |" formula to concatenate items from @p errors
+// is used.
+// @param errors A vector of error strings.
+// @return An empty string when @p errors is empty, otherwise a string with all the items in @p errors.
+std::string AggregateRouteConnectivityErrors(const std::vector<std::string>& errors) {
+  std::string result{errors.empty() ? "" : "Route has connectivity errors: "};
+  for (const std::string& error : errors) {
+    result += error + " | ";
+  }
+  return result;
+}
+
+// Iterates over @p routes and validates none have end to end connectivity errors.
+// @param routes Vector of Routes to validate.
+// @throws common::assertion_error When one of the Routes in @p routes has end to end connectivity errors.
+void ValidateEndToEndConnectivityInRoutes(const std::vector<routing::Route>& routes) {
+  for (const auto& route : routes) {
+    const std::vector<std::string> connectivity_errors = route.ValidateEndToEndConnectivity();
+    MALIPUT_VALIDATE(connectivity_errors.empty(), AggregateRouteConnectivityErrors(connectivity_errors));
+  }
+}
+
+}  // namespace
+
+DistanceRouter::DistanceRouter(const api::RoadNetwork& road_network, double lane_s_range_tolerance)
+    : Router(), road_network_(road_network), lane_s_range_tolerance_(lane_s_range_tolerance) {
+  MALIPUT_THROW_UNLESS(lane_s_range_tolerance_ >= 0.0);
+}
+
+std::vector<routing::Route> DistanceRouter::DoComputeRoutes(
+    const api::RoadPosition& start, const api::RoadPosition& end,
+    const routing::RoutingConstraints& routing_constraints) const {
+  // Validate input parameters.
+  MALIPUT_THROW_UNLESS(start.lane != nullptr);
+  MALIPUT_THROW_UNLESS(road_network_.road_geometry()->ById().GetLane(start.lane->id()) == start.lane);
+  MALIPUT_THROW_UNLESS(end.lane != nullptr);
+  MALIPUT_THROW_UNLESS(road_network_.road_geometry()->ById().GetLane(end.lane->id()) == end.lane);
+  routing::ValidateRoutingConstraints(routing_constraints);
+  static constexpr bool kRemoveUTurns{true};
+
+  // Obtain the lane sequences that connect the start with end, i.e. the routing algorithm.
+  const std::vector<std::vector<const api::Lane*>> lane_sequences =
+      routing::FindLaneSequences(start.lane, end.lane, std::numeric_limits<double>::max(), kRemoveUTurns);
+
+  // Construct the routing::Routes and routing::Phases.
+  // TODO: lateral routing::Phase inflation.
```

Comment:
It means to consider other lanes in the segment besides the one that FindLaneSequences() can return.
I expect to change the code here in a follow up PR integrating #635 contents such that instead of routing with lanes, the router can use segments and then "inflate" phases of routes.

Reply:
I see, so it is lateral inflation since any alternative lane from within the same segment will result in lateral movement and a longer distance traveled? Would the following comment be clearer?

```
  // TODO: Add support for lateral routing::Phase inflation by considering alternative lanes from within the same
  // segment besides those returned by routing::FindLaneSequences().
```

Reply:
Fixed in [56c49b3](https://github.com/maliput/maliput/pull/619/commits/56c49b3f0396d03683c9ebc9015ea2f26539b1a1)